common/rabbitmq/rabbitmq.go

```diff
@@ -42,7 +44,9 @@ func PublishNotification(channel *amqp.Channel, body *NotificationBody) error {
 	if err != nil {
 		return err
 	}
-	return channel.Publish("", NOTIFICATION_QUEUE_NAME, false, false, amqp.Publishing{
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
`ctx` should be an argument, so that the caller itself can define canceling behavior. This is a common pattern in Go.

Reply:
Done